wdl/CallSmallVariants.wdl

```diff
@@ -0,0 +1,99 @@
+version 1.0
+
+import "Structs.wdl"
+
+task DeepVariant {
+    input {
+        File bam
+        File bai
+
+        File ref_fasta
+        File ref_fai
+
+        String model_class
+
+        String output_prefix
+
+        String? intervals
+
+        RuntimeAttr? runtime_attr_override
+    }
+
+    Int disk_size = ceil(size(bam, "GB")) + 50
+
+    String extra_args = if defined(intervals) then "--regions=~{intervals}" else " "
+
+    command <<<
+
+        # example from https://github.com/google/deepvariant/blob/r0.8/docs/deepvariant-quick-start.md
+        set -euo pipefail
+
+        # this is just to future proof ourselves for ONT data
+        if [[ ~{model_class} != "PACBIO" ]]; then
+          echo "requesting a model type not supported by the model: ~{model_class}"
+          exit 1
+        fi
+
+        num_core=$(cat /proc/cpuinfo | awk '/^processor/{print $3}' | wc -l)
+
+        /opt/deepvariant/bin/run_deepvariant \
+          --model_type=~{model_class} \
+          --ref=~{ref_fasta} \
+          --reads=~{bam} \
+          --output_vcf=~{output_prefix}.vcf.gz \
+          --output_gvcf=~{output_prefix}.g.vcf.gz \
+          --num_shards=${num_core} \
+          ~{extra_args}
+    >>>
+
+    output {
+        # save both VCF and gVCF
+        File gvcf = "~{output_prefix}.deepvariant.g.vcf.gz"
+        File vcf = "~{output_prefix}.deepvariant.vcf.gz"
+    }
+
+    #########################
+    RuntimeAttr default_attr = object {
+        cpu_cores:          8, 
+        mem_gb:             50, 
+        disk_gb:            "~{disk_size}",
+        boot_disk_gb:       10,
+        preemptible_tries:  1,
+        max_retries:        0,
+        docker:             "gcr.io/deepvariant-docker/deepvariant@sha256:699870a8c4109c13f59a9a5899db563011746d6104aa78c999cbcc715d03dcf2"
+    }
+    RuntimeAttr runtime_attr = select_first([runtime_attr_override, default_attr])
+    runtime {
+        cpu:                    select_first([runtime_attr.cpu_cores,         default_attr.cpu_cores])
+        memory:                 select_first([runtime_attr.mem_gb,            default_attr.mem_gb]) + " GiB"
+        disks: "local-disk " +  select_first([runtime_attr.disk_gb,           default_attr.disk_gb]) + " HDD"
+        bootDiskSizeGb:         select_first([runtime_attr.boot_disk_gb,      default_attr.boot_disk_gb])
+        preemptible:            select_first([runtime_attr.preemptible_tries, default_attr.preemptible_tries])
+        maxRetries:             select_first([runtime_attr.max_retries,       default_attr.max_retries])
+        docker:                 select_first([runtime_attr.docker,            default_attr.docker])
+    }
+}
+
+workflow TestCallSmallVariants {
```

Comment:
Question - if a WDL has tasks and a workflow defined, can a different WDL import the file and just run the tasks?

Reply:
Yes.

[This workflow](https://github.com/broadinstitute/dsde-pipelines/blob/master/pipelines/dna_seq/germline/wgs/WholeGenomeGermlineSingleSample.wdl#L31) imports [another WDL](https://github.com/broadinstitute/dsde-pipelines/blob/master/pipelines/dna_seq/UnmappedBamToAlignedBam.wdl), both having their own workflows defined.
The first apparently [calls](https://github.com/broadinstitute/dsde-pipelines/blob/master/pipelines/dna_seq/germline/wgs/WholeGenomeGermlineSingleSample.wdl#L59) the `workflow` defined in the second.

Reply:
That's super useful - in the (near) future, each one of our sub-WDLs could have their own workflows so that we can run individual components without having to rerun the entire pipeline.  Cool!